op-node: Hide rethDB option

trianglesphere opened the pull request:
**Description**

If the rethDB option is used without the proper build configuration, the op-node will panic. Note that we do not have this build configuration in our releases. This hides the option to use it, but does not fully remove it. There was some duplication required to easily hide the command line flag while still being able to use it for testing purposes.

This commit should be reverted when rethDB is stable.

**Tests**

Tested flags localy.

sebastianst commented on op-service/sources/receipts.go:
```diff
@@ -138,6 +138,20 @@ var RPCProviderKinds = []RPCProviderKind{
+}
+
+// Copy of RPCProviderKinds with RethDB added to all RethDB to be used but to hide it from the flags
+var validRPCProviderKinds = []RPCProviderKind{
+	RPCKindAlchemy,
+	RPCKindQuickNode,
+	RPCKindInfura,
+	RPCKindParity,
+	RPCKindNethermind,
+	RPCKindDebugGeth,
+	RPCKindErigon,
+	RPCKindBasic,
+	RPCKindAny,
+	RPCKindStandard,
 	RPCKindRethDB,
 }
```
could do something like
```suggestion
var validRPCProviderKinds = func() []RPCProviderKind{
    return append(RPCProviderKinds, RPCKindRethDB)
}()
```
to avoid duplication.

trianglesphere committed:
op-node: Hide rethDB option

If the rethDB option is used without the proper build configuration,
the op-node will panic. Note that we do not have this build configuration
in our releases. This hides the option to use it, but does not fully
remove it. There was some duplication required to easily hide the
command line flag while still being able to use it for testing purposes.

This commit should be reverted when rethDB is stable.
